src/app/public/modules/auth-http-client/auth-interceptor.ts

```diff
@@ -125,4 +126,37 @@ export class SkyAuthInterceptor implements HttpInterceptor {
     return next.handle(request);
   }
 
+  public getClient(handler: HttpHandler): HttpClient {
```

Comment:
can this be private?

Reply:
i see now why it's public. you're trying to make this testable by allowing the client to be mocked.

couple thoughts on that, none of which i'm 100% committed to:
- you could mock the `HttpHandler` instead. `HttpHandler` is an abstract class that you could extend to do what you want in the test and forego a mock altogether. either way, the client calls [`.handle`](https://github.com/angular/angular/blob/master/packages/common/http/src/client.ts#L500) so you could mock that in the test.
- you could move the logic of checking the `/version` into a separte service class (`LocalEndpointChecker` or something) and let that be a standard dependency via the constructor of the interceptor. (it'd probably have to be a `@Optional` dep for backwards compatibility sake.) then you can use the standard angular DI to inject your mock one.

Reply:
I can look into that.  This was the quickest thing I could think of before getting PRs up.